test/integ/pandas/test_arrow_pandas.py

```diff
@@ -1169,6 +1169,16 @@ def assert_dtype_equal(a, b):
     )
 
 
+def assert_pandas_batch_types(batch, expected_types):
+    assert batch.dtypes is not None
+
+    pandas_dtypes = batch.dtypes
+    # pd.string is represented as an np.object
+    # np.dtype string is not the same as pd.string (python)
+    for i, typ in enumerate(expected_types):
+        assert_dtype_equal(pandas_dtypes[i].type, numpy.dtype(typ).type)
```

Comment:
I think `zip` would work better here:
```suggestion
    for a, b in zip(pandas_dtypes, expected_types):
        assert_dtype_equal(a.type, numpy.dtype(b).type))
```